test/unstoppable/README.md

```diff
@@ -1,365 +1,308 @@
+halmos_target_address_dc9e083_01 = 0x00000000000000000000000000000000aaaa0005
+```
+Given that we already know that **0x0...aaaa0005** is the attacker's address, it is easy to guess that we are dealing with recursion. Attacker calls its own attack() function and thus inflates the analysis with garbage counterexamples. Fixing this is quite simple - just use the vm.assume() cheat code when creating a symbolic address:
+```javascript
+function attack() public {
+        address target = svm.createAddress("target");
+        vm.assume (target != address(this)); // Avoid recursion
+        ...
 ```
-SPOILER: this is exactly what we are searching for!
+Now we've gotten rid of the trashy recursive counterexamples and have only meaningful one.
+
+### Calldata analysis
 
-Let's break down this output:
-```bash
-halmos_contract_index_uint256_01 = 0x0000000000000000000000000000000000000000000000000000000000000000 (0)
+We already know that the attack is based on some transaction to the token address. Let's break down this calldata brick by brick:
+```javascript
+0xa9059cbb00000000000000000000000000000000000000000000000000000000aaaa0003000000000000000000000000000000000000000000000000743133125f0000010000000000000000000000000000000000000000000000000000000000000000
 ```
-means that this is the 0th contract from known contracts list (DamnValuableToken contract)
-```bash
-halmos_data_bytes_02 = 0xa9059cbb00000000000000000000000000000000000000000000000000000000aaaa000300000000000000000000000000000000000000000000000003c723168ff988000000000000000000000000000000000000000000000000000000000000000000 (100 bytes)
+The first 4 bytes (a9059cbb) are the selector of some token function. It is easy to find out which selector it is. It is enough to look at the compiler artifacts in the file **out/DamnValuableToken.sol/DamnValuableToken.json** and enter this selector in the search:
+```json
+...,"transfer(address,uint256)":"a9059cbb",...
 ```
-is more interesting. First 4 bytes "a9059cbb" are token "transfer" selector
-Next bytes 
-```bash
+This is just a transfer function.
+The parameters of this function are also simple:
+```javascript
 00000000000000000000000000000000000000000000000000000000aaaa0003
 ```
-are some address. This is tokens receiver address. We'll find out what address is it later.
-And the third part 
-```bash
-00000000000000000000000000000000000000000000000003c723168ff988000000000000000000000000000000000000000000000000000000000000000000
-```
-is amount of tokens to transfer. 0s in the end is padding. Since we generated 100 symbolic bytes, it generates 0s padding. We can leave it as is.
-Now, it is necessary to find out what the address is our "receiver". We'll use console logging functional from forge and print all contracts addresses in "setUp()" function:
+Obviously, this is the address of the vault contract.
+The remaining bytes are the amount that we pass to the transfer function:
```

Comment:
can also add instructions for decoding the abi with `cast` instead of manually:

```sh
$ cast abi-decode --input "transfer(address,uint256)()" 00000000000000000000000000000000000000000000000000000000aaaa0003000000000000000000000000000000000000000000000000743133125f0000010000000000000000000000000000000000000000000000000000000000000000

0x00000000000000000000000000000000aaaA0003
8372529336254726145 [8.372e18]
```